src/json/to_json.cr

```diff
@@ -61,7 +61,7 @@ struct Float
 end
 
 class String
-  def to_json(json : JSON::Builder) : IO
+  def to_json(json : JSON::Builder) : Nil
```

Comment:
Righto.... so it's this change that caused a bug to appear in the doc generator

Reply:
Yeah, it's really just a bug in the doc generator.

https://github.com/crystal-lang/crystal/blob/master/src/compiler/crystal/tools/doc/type.cr#L763

It should be `superclass ? superclass.to_json_simple(builder) || builder.null`

Reply:
Right. So I pushed the fix.
It's a bit concerning that there's already a bug in stdlib itself, so what to say about other code...
On the other hand, maybe there aren't even such extensive usages of JSON::Builder

Reply:
Well, at least this change helps identify such bugs.